ledger/participant-integration-api/src/test/suite/scala/platform/store/CompletionFromTransactionSpec.scala

```diff
@@ -3,39 +3,102 @@
 
 package com.daml.platform.store
 
+import com.daml.ledger.api.v1.completion.Completion.DeduplicationPeriod
 import com.daml.ledger.offset.Offset
+import com.google.protobuf.duration.Duration
 import com.google.protobuf.timestamp.Timestamp
 import com.google.rpc.status.Status
 import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
 import org.scalatest.wordspec.AnyWordSpec
 
 import java.time.Instant
 
-class CompletionFromTransactionSpec extends AnyWordSpec with Matchers {
+class CompletionFromTransactionSpec
+    extends AnyWordSpec
+    with Matchers
+    with TableDrivenPropertyChecks {
 
   "CompletionFromTransaction" should {
-    "acceptedCompletion" in {
-      val completionStream = CompletionFromTransaction.acceptedCompletion(
-        Instant.EPOCH,
-        Offset.beforeBegin,
-        "commandId",
-        "transactionId",
-        "applicationId",
-        Some("submissionId"),
+    "create an accepted completion" in {
+      val testCases = Table(
+        (
+          "deduplicationOffset",
+          "deduplicationDurationSeconds",
+          "deduplicationDurationNanos",
+          "expectedDeduplicationPeriod",
+        ),
+        (None, None, None, DeduplicationPeriod.Empty),
+        (Some("offset"), None, None, DeduplicationPeriod.DeduplicationOffset("offset")),
+        (
+          None,
+          Some(1L),
+          Some(2),
+          DeduplicationPeriod
+            .DeduplicationDuration(new Duration(1, 2))
+            .asInstanceOf[ // otherwise the compilation fails due to an inference warning
+              DeduplicationPeriod
+            ],
+        ),
       )
 
-      val checkpoint = completionStream.checkpoint.get
-      checkpoint.recordTime shouldBe Some(Timestamp(Instant.EPOCH))
-      checkpoint.offset shouldBe a[Some[_]]
+      forEvery(testCases) {
+        (
+            deduplicationOffset,
+            deduplicationDurationSeconds,
+            deduplicationDurationNanos,
+            expectedDeduplicationPeriod,
+        ) =>
+          val completionStream = CompletionFromTransaction.acceptedCompletion(
+            Instant.EPOCH,
+            Offset.beforeBegin,
+            "commandId",
+            "transactionId",
+            "applicationId",
+            Some("submissionId"),
```

Comment:
Can you also check an absent submission ID?

Reply:
Done